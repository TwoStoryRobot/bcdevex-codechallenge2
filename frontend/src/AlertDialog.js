import React from 'react'
import PropTypes from 'prop-types'

import Dialog from '@material-ui/core/Dialog'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogActions from '@material-ui/core/DialogActions'
import Button from '@material-ui/core/Button'

function AlertDialog({ children, onCancel, onOk, ...props }) {
  return (
    <Dialog {...props}>
      <DialogContent>
        <DialogContentText>{children}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onCancel}>Cancel</Button>
        <Button onClick={onOk}>Ok</Button>
      </DialogActions>
    </Dialog>
  )
}

AlertDialog.propTypes = {
  children: PropTypes.text
}

AlertDialog.defaultProps = {
  children: 'Are you sure you want to do this?'
}

export default AlertDialog
