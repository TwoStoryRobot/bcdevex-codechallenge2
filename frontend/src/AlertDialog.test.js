import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'

import AlertDialog from './AlertDialog'

afterEach(cleanup)

test('should render with required props', async () => {
  const { getByTestId } = render(
    <AlertDialog open={true} onCancel={jest.fn()} onConfirm={jest.fn()}>
      Are you TESTING?
    </AlertDialog>
  )

  // AlertDialog wraps Dialog which is a modal and renders using a portal
  // must get by testid to find the element in the document
  expect(getByTestId('alert-dialog')).toMatchSnapshot()
})

test('should call onCancel when labeled cancel clicked', async () => {
  const mockCancel = jest.fn()
  const { getByText } = render(
    <AlertDialog
      open={true}
      cancelLabel="TESTCANCEL"
      onCancel={mockCancel}
      onConfirm={jest.fn()}
    />
  )

  fireEvent.click(getByText('TESTCANCEL'))

  expect(mockCancel).toHaveBeenCalled()
})

test('should call onConfirm when labeled confirm clicked', async () => {
  const mockConfirm = jest.fn()
  const { getByText } = render(
    <AlertDialog
      open={true}
      onCancel={jest.fn()}
      confirmLabel="TESTCONFIRM"
      onConfirm={mockConfirm}
    />
  )

  fireEvent.click(getByText('TESTCONFIRM'))

  expect(mockConfirm).toHaveBeenCalled()
})
