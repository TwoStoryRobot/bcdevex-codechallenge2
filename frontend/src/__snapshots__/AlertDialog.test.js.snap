// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with required props 1`] = `
<div
  class="MuiModal-root-013 MuiDialog-root-01 MuiDialog-scrollPaper-02"
  role="dialog"
  data-testid="alert-dialog"
>
  <div
    class="MuiBackdrop-root-015"
    aria-hidden="true"
    style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    class="MuiPaper-root-017 MuiPaper-elevation24-043 MuiPaper-rounded-018 MuiDialog-paper-04 MuiDialog-paperScrollPaper-05 MuiDialog-paperWidthSm-08"
    style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    role="document"
    tabindex="-1"
  >
    <div
      class="MuiDialogContent-root-044"
    >
      <p
        class="MuiTypography-root-046 MuiDialogContentText-root-045 MuiTypography-subheading-053 MuiTypography-colorTextSecondary-069"
      >
        Are you TESTING?
      </p>
    </div>
    <div
      class="MuiDialogActions-root-071"
    >
      <button
        tabindex="0"
        class="MuiButtonBase-root-099 MuiButton-root-073 MuiButton-text-075 MuiButton-textPrimary-076 MuiButton-flat-078 MuiButton-flatPrimary-079 MuiDialogActions-action-072"
        type="button"
      >
        <span
          class="MuiButton-label-074"
        >
          Cancel
        </span>
        <span
          class="MuiTouchRipple-root-0102"
        />
      </button>
      <button
        tabindex="0"
        class="MuiButtonBase-root-099 MuiButton-root-073 MuiButton-text-075 MuiButton-textPrimary-076 MuiButton-flat-078 MuiButton-flatPrimary-079 MuiDialogActions-action-072"
        type="button"
      >
        <span
          class="MuiButton-label-074"
        >
          Confirm
        </span>
        <span
          class="MuiTouchRipple-root-0102"
        />
      </button>
    </div>
  </div>
</div>
`;
